client/src/components/dataList/SpeciesInfo.js

```diff
@@ -0,0 +1,293 @@
+
+        case 'editPopulation':
+            return { ...state, population: action.payload };
+
+        case 'editConservationStatus':
+            return { ...state, conservation_status: action.payload };
+        case 'editCreatedOn':
+            return { ...state, created_on: action.payload };
+        case 'clearForm':
+            return { id: "", common_name: '', scientific_name: '', population: '', conservation_status: '', created_on: '' };
+        default:
+            return state;
+    }
+};
+const SpeciesInfo = () => {
+    const [species, setSpecies] = useState([]);
+
+    //get species data table
+    const getSpecies = async () => {
+        const response = await fetch(
+            `http://localhost:4040/species`
+        );
+        const specie = await response.json();
+        console.log(specie);
+        setSpecies(specie);
+    };
+
+    useEffect(() => {
+        getSpecies();
+    }, []);
+
+    //initialistate of the form will be empty
+    const initialState = {
+        id: "",
+        common_name: '',
+        scientific_name: '',
+        population: '',
+        conservation_status: '',
+        created_on: ''
+    };
+
+    const [state, dispatch] = useReducer(reducer, initialState);
+    console.log(state);
+
+    const handleAddSpecie = async (e) => {
+        e.preventDefault();
+
+        const newSpecie = {
+            id: state.id,
+            common_name: state.common_name,
+            scientific_name: state.scientific_name,
+            population: state.population,
+            conservation_status: state.conservation_status,
+            created_on: state.created_on
+        }
+        console.log(newSpecie);
+        const response = await fetch('http://localhost:4040/species', {
+            method: 'POST',
+            headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(newSpecie)
+        });
+        const content = await response.json();
+        setSpecies([...species, content])
+        dispatch({ type: 'clearForm' });
+    }
+
+    //delete specie handler
+    const handleDeleteSpecie = async (deleteId) => {
+        const response = await fetch(`http://localhost:4040/species/${deleteId}`, {
+            method: 'DELETE',
+        })
+        await response.json();
+        const deleteSpecieFunction = species.filter((specie) => specie.id !== deleteId);
+        setSpecies(deleteSpecieFunction);
+    };
+    const [searchTerm, setSearchTerm] = useState('');
+
+    const [editSpecieId, setEditSpecieId] = useState(null);
+
+    const addSpecie = (newSpecie) => {
+        setSpecies((species) => [...species, newSpecie])
+    }
+    const updateSpecie = (savedSpecie) => {
+        console.log("Line 29 savedStudent", savedSpecie);
+        // This function should update the whole list of students - 
+        setSpecies((species) => {
```

Comment:
I'd  probably call this `originalSpeciesList` to make it clear 1) it is a list 2) it is the original list the fe has.